### DISCUSSION TITLE
Add Support for Tab Groups

### DISCUSSION THREAD

**Discussion by wheelieman711** [sienori/Tab-Session-Manager#705] [Ideas]
I think it would be great if you could expand the functionality of this extension to work with tab groups in Google Chrome.

**Comment by Newtomic** [comment]
Yes please! That would be great.

**Comment by sienori** [comment]
In version 6.6.0, I implemented support for tab groups! https://github.com/sienori/Tab-Session-Manager/commit/f3e14f7d30ec9e46a58d6a84848f6a4b677a7591 #652
To save tab groups, you need to install [an additional extension](https://github.com/sienori/Save-Tab-Groups-for-Tab-Session-Manager).
![image](https://user-images.githubusercontent.com/29514424/116112870-f35a2600-a6f2-11eb-8454-0ad5e4c9da96.png)

**Reply by wheelieman711** [reply]
Awesome! It seems to work great for Brave Browser as well which I recently started using for privacy reasons.

**Reply by Newtomic** [reply]
> In version 6.6.0, I implemented support for tab groups! [f3e14f7](https://github.com/sienori/Tab-Session-Manager/commit/f3e14f7d30ec9e46a58d6a84848f6a4b677a7591) #652
> To save tab groups, you need to install [an additional extension](https://github.com/sienori/Save-Tab-Groups-for-Tab-Session-Manager).
> ![image](https://user-images.githubusercontent.com/29514424/116112870-f35a2600-a6f2-11eb-8454-0ad5e4c9da96.png)

It work's great but an additional extension why?

**Reply by lbecque** [reply]
We have tab groups in Firefox now.  Does this also work for FF?  I agree that this should be incorporated into the main code of tab session manager rather than be separate extension.

**Reply by Haiyami** [reply]
> We have tab groups in Firefox now. Does this also work for FF? I agree that this should be incorporated into the main code of tab session manager rather than be separate extension.

It's only for chrome or chromium based browsers so far. This hasn't been applied to the firefox version of the extension. Though it used to work in Waterfox before they removed the ability to add Chrome extensions from the browser code. This is why the Firefox version needs this feature as well.

**Reply by lbecque** [reply]
In the latest 2026 update it looks like tab groups functionality has been added to Firefox.  I tested and its working great!  Thanks @sienori !

**Comment by DRYN07** [comment]
It would be awesome to make this work with Vivaldi Browser

**Comment by Haiyami** [comment]
This is only available for chrome. Why hasn't saving tab groups been added for Firefox yet?

**Reply by lbecque** [reply]
its working for me now in 2026
